### PR TITLE
doc: net: Add networking changes to 2.5 release note

### PR DESCRIPTION
**Pull request by jukkar**
Initial set of networking related changes in 2.5 release.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: doc/releases/release-notes-2.5.rst [hunk 1]
```diff
@@ -376,12 +423,139 @@ Drivers and Sensors
...
+    still available and scheduled for removal in next 2.6 release.
+  * Added support to queue received out-of-order TCP data.
+  * Added connection termination if the TCP handshake is not finalized.
+  * Enhanced received TCP RST packet handling.
+  * Fixed TCP connection from Windows 10.
+
+* TLS:
+
+  * Use Maximum Fragment Length (MFL) extension by default.
+  * Added ALPN extension option to TLS.
```
Comment (mniestroj) [inline doc/releases/release-notes-2.5.rst:558]: Also:
- Fixed TLS context leak on socket allocation failure.

File: doc/releases/release-notes-2.5.rst [hunk 2]
```diff
@@ -343,6 +358,38 @@ Drivers and Sensors
 
 * Modem
 
+  * Improved RX with HW flow control in modem interface API.
+  * Improved reading from interface in command handler.
+  * Fixed race condition when waiting on cmd reply.
+  * Added support for Quectel bg95 modem.
```
Comment (mniestroj) [inline doc/releases/release-notes-2.5.rst:364]: Also:
- Constified modem command structures to reduce RAM usage.

**Review by nashif — APPROVED** [review]

File: doc/releases/release-notes-2.5.rst [hunk 3]
```diff
@@ -376,12 +423,139 @@ Drivers and Sensors
 
 * WiFi
 
-  * Added uart bus interface for eswifi driver.
+  * eswifi:
+
+    * Added uart bus interface. This enables all Inventek modules with
+      IWIN AT Commands firmware.
+
+  * esp32:
```
Comment (mniestroj) [inline doc/releases/release-notes-2.5.rst:431]: `s/esp32/esp/` This is a driver for both ESP32 and ESP8266 (and some variants of those).

**Review by mniestroj — APPROVED** [review]

**Review by nandojve — APPROVED** [review]
Question: Is NCP still supported?

**Comment by jukkar** [comment]
> Is NCP still supported?

Seems to be (from `subsys/net/l2/openthread/Kconfig`)
```
choice
        prompt "OpenThread Co-Processor type"
        help
          This option selects Thread network co-processor type

config OPENTHREAD_COPROCESSOR_NCP
        bool "NCP - Network Co-Processor"
config OPENTHREAD_COPROCESSOR_RCP
        bool "RCP - Radio Co-Processor"
endchoice
```

**Pull request merged**